File: src/scraper/entities/weapon.entity.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { WeaponLevelRank } from './weapon_level_rank.entity';

@Entity()
export class WeaponEntity {

  @PrimaryGeneratedColumn()
  id: number;

  @Column({ unique: true })
  name: string;

  @Column({ unique: true })
  href: string;

  @Column()
  rank: number;

  @Column()
  type: string;

  @Column()
  imageUrl: string;

  @OneToMany(() => WeaponLevelRank, levelRank => levelRank.weapon)
  levelRanks: WeaponLevelRank[];

}

File: mobile/lib/domains/weapon/weapon_entity.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:mobile/core/enums/enums.dart';
import 'package:mobile/domains/weapon/weapon_level_rank_entity.dart';

part 'weapon_entity.freezed.dart';
part 'weapon_entity.g.dart';
@freezed
class WeaponEntity with _$WeaponEntity {
  const factory WeaponEntity({
    int? id,
    String? name,
    String? href,
    int? rank,
    WeaponType? type,
    String? imageUrl,
    @JsonKey(name: 'levelRanks') List<WeaponLevelRankEntity>? weaponLevelRank,

  }) = _WeaponEntity;

  factory WeaponEntity.fromJson(Map<String, dynamic> json) =>
      _$WeaponEntityFromJson(json);
}